examples/monaco.js

```diff
@@ -32,22 +31,59 @@ const editors = [
   "extending-language-services-configure-json-defaults",
 ];
 
-(async () => {
-  const browser = await firefox.launch();
-  const context = await browser.newContext();
+const selectors = {
+  editor: `.monaco-editor`,
+  get input() {
+    return `${this.editor} textarea.inputarea`;
+  },
+  get lines() {
+    return `${this.editor} .view-line`;
+  },
+  tabs: ".tabArea",
+  get run() {
+    return `${this.tabs} button`;
+  },
+  tabByText(text) {
+    return `${this.tabs} .tab >> :text="${text}"`;
+  },
+  select: ".sample-switcher",
+  runner: "#runner",
+  runnerContainer: "#container",
+};
 
-  const page = await context.newPage();
+const clearInput = async (page) => {
+  await page.click(selectors.editor);
 
+  await page.press(selectors.input, "Meta+A");
+  await page.press(selectors.input, "Backspace");
+};
+
+example("Monaco Editor", async (page, { action }) => {
   await page.goto("https://microsoft.github.io/monaco-editor/playground.html");
 
-  for (const editor of editors) {
-    console.log(editor);
-    await page.selectOption(".sample-switcher", editor);
-    await page.waitForTimeout(500);
-  }
+  await action("Iterate samples", async (page, { log }) => {
+    for (const editor of editors) {
+      await page.selectOption(selectors.select, editor);
+      await page.waitForTimeout(1000);
+    }
```

Comment:
haha we can always trim this back to ~5 or so editors if we need to

Reply:
Sure. My goal right now is to get everything running and reporting consistently and then revisit with an eye for the purpose of the test. The exercise of updating the tests has been helpful to acclimate to the environment and develop some reasonable abstractions.